src/code/RepositorySettings.cs

```diff
@@ -74,20 +74,178 @@ public static void CheckRepositoryStore()
+        {
+            errorMsgs = Utils.EmptyStrArray;
+            List<string> tempErrors = new List<string>();
+            // remove trailing and leading whitespaces, and if Name is just whitespace Name should become null now and be caught by following condition
+            repoName = repoName.Trim(' ');
+            if (String.IsNullOrEmpty(repoName) || repoName.Contains("*"))
+            {
+                throw new ArgumentException("Name cannot be null/empty, contain asterisk or be just whitespace");
+            }
+
+            if (repoUri == null || !(repoUri.Scheme == System.Uri.UriSchemeHttp || repoUri.Scheme == System.Uri.UriSchemeHttps || repoUri.Scheme == System.Uri.UriSchemeFtp || repoUri.Scheme == System.Uri.UriSchemeFile))
+            {
+                throw new ArgumentException("Invalid Uri, must be one of the following Uri schemes: HTTPS, HTTP, FTP, File Based");
+            }
+
+            if (repoCredentialInfo != null)
+            {
+                bool isSecretManagementModuleAvailable = Utils.IsSecretManagementModuleAvailable(repoName, cmdletPassedIn);
+
+                if (repoCredentialInfo.Credential != null)
+                {
+                    if (!isSecretManagementModuleAvailable)
+                    {
+                        tempErrors.Add($"Microsoft.PowerShell.SecretManagement module is not found, but is required for saving PSResourceRepository {repoName}'s Credential in a vault.");
+                    }
+                    else
+                    {
+                        Utils.SaveRepositoryCredentialToSecretManagementVault(repoName, repoCredentialInfo, cmdletPassedIn);
+                    }
+                }
+
+                if (!isSecretManagementModuleAvailable)
+                {
+                    cmdletPassedIn.WriteWarning($"Microsoft.PowerShell.SecretManagement module cannot be found. Make sure it is installed before performing PSResource operations in order to successfully authenticate to PSResourceRepository \"{repoName}\" with its CredentialInfo.");
+                }
+            }
+
+            errorMsgs = tempErrors.ToArray();
+            if (!cmdletPassedIn.ShouldProcess(repoName, "Register repository to repository store"))
+            {
+                return null;
+            }
+
+            if (errorMsgs.Length > 0)
+            {
+                return null;
+            }
+
+            var repo = RepositorySettings.Add(repoName, repoUri, repoPriority, repoTrusted, repoCredentialInfo, force);
+
+            return repo;
+        }
+
+
+        public static PSRepositoryInfo UpdateRepositoryStore(string repoName, Uri repoUri, int repoPriority, bool repoTrusted, bool isSet, int defaultPriority, PSCredentialInfo repoCredentialInfo, PSCmdlet cmdletPassedIn, out string[] errorMsgs)
+        {
+            var tempErrors = new List<string>() { };
```

Comment:
I believe you don't need bracket initializer
```suggestion
            var tempErrors = new List<string>();
```